Source/JavaScriptCore/bytecode/InlineCacheCompiler.cpp

```diff
@@ -643,6 +649,489 @@ ScratchRegisterAllocator InlineCacheCompiler::makeDefaultScratchAllocator(GPRReg
     return allocator;
 }
 
+#if CPU(X86_64) && OS(WINDOWS)
+static constexpr size_t prologueSizeInBytesDataIC = 4;
+#elif CPU(X86_64)
+static constexpr size_t prologueSizeInBytesDataIC = 1;
+#elif CPU(ARM64E)
+static constexpr size_t prologueSizeInBytesDataIC = 8;
+#elif CPU(ARM64)
+static constexpr size_t prologueSizeInBytesDataIC = 4;
+#elif CPU(ARM_THUMB2)
+static constexpr size_t prologueSizeInBytesDataIC = 4;
+#elif CPU(MIPS)
+static constexpr size_t prologueSizeInBytesDataIC = 16;
+#elif CPU(RISCV64)
+static constexpr size_t prologueSizeInBytesDataIC = 12;
+#else
+#error "unsupported architecture"
+#endif
+
+void InlineCacheCompiler::emitDataICPrologue(CCallHelpers& jit)
+{
+    // Important difference from the normal emitPrologue is that DataIC handler does not change callFrameRegister.
+    // callFrameRegiste is an original one of the caller JS function. This removes necessity of complicated handling
```

Comment:
typo: callFrameRegiste. It should probably also be "is the original one from the caller JS function"

Reply:
Fixed.